pkg/kubelet/lifecycle/handlers.go

```diff
@@ -121,15 +151,30 @@ func (hr *handlerRunner) runHTTPHandler(pod *v1.Pod, container *v1.Container, ha
 	var port int
 	if handler.HTTPGet.Port.Type == intstr.String && len(handler.HTTPGet.Port.StrVal) == 0 {
 		port = 80
+		if utilfeature.DefaultFeatureGate.Enabled(features.LifecycleHandlerHTTPS) && handler.HTTPGet.Scheme == v1.URISchemeHTTPS {
+			port = 443
+		}
 	} else {
 		var err error
 		port, err = resolvePort(handler.HTTPGet.Port, container)
 		if err != nil {
 			return "", err
 		}
 	}
-	url := fmt.Sprintf("http://%s/%s", net.JoinHostPort(host, strconv.Itoa(port)), handler.HTTPGet.Path)
-	resp, err := hr.httpGetter.Get(url)
+	path := handler.HTTPGet.Path
+	scheme := "http"
+	if utilfeature.DefaultFeatureGate.Enabled(features.LifecycleHandlerHTTPS) && handler.HTTPGet.Scheme == v1.URISchemeHTTPS {
+		scheme = "https"
+	}
+
+	url := formatURL(scheme, host, port, path)
+	req, err := http.NewRequest(http.MethodGet, url.String(), nil)
+	if err != nil {
+		return "", err
+	}
+	req.Header = buildHeader(handler.HTTPGet.HTTPHeaders)
+	resp, err := hr.httpDoer.Do(req)
```

Comment:
for compatibility, should this fall back to http if the target does not support TLS?

Reply:
it seems to do it "automagically" 
```go
package main

import (
        "crypto/tls"
        "fmt"
        "io/ioutil"
        "log"
        "net/http"
)

func main() {

        http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
                fmt.Fprintf(w, "Hello")
        })
        log.Println("create server")

        go func() {
                log.Fatal(http.ListenAndServe("localhost:8081", nil))
        }()

        client := &http.Client{
                Transport: &http.Transport{
                        TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
                },
        }
        log.Println("request")
        req, err := http.NewRequest("GET", "http://localhost:8081", nil)
        if err != nil {
                panic(err)
        }

        response, err := client.Do(req)
        if err != nil {
                panic(err)
        }

        defer response.Body.Close()
        body, err := ioutil.ReadAll(response.Body)
        if err != nil {
                panic(err)
        }

        fmt.Println("Response", string(body))
}
```

Reply:
@aojea
``` go
req, err := http.NewRequest("GET", "https://localhost:8081", nil)
```
It should start with `https://` for TLS

Reply:
ah, great, mystery solved then, there is no auto fallback :sweat_smile: